docs/guides/tms.rst

```diff
@@ -0,0 +1,252 @@
+TMS Servers
+===========
+
+GeoPySpark is meant to work with geospatial data.  The most natural way to
+interact with these data is to display them on a map.  In order to allow for
+this interactive visualization, we provide a means to create Tile Map Service
+(TMS) servers directly from both GeoPySpark RDDs and tile catalogs.  A TMS
+server may be viewed using a web-based tool such as geojson.io_ or interacted
+with using the GeoNotebook_ Jupyter kernel. [#]_
+
+Note that the following examples rely on this common boilerplate code:
+
+.. code-block:: python
+
+   import geopyspark as gps
+   from pyspark import SparkContext
+
+   conf = gps.geopyspark_conf(appName="demo")
+   sc = SparkContext(conf=conf)
+
+Basic Example
+-------------
+
+The most straightforward use case of the TMS server is to display a singleband
+layer with some custom color map.  This is accomplished easily:
+
+.. code-block:: python
+
+   cm = gps.ColorMap.nlcd_colormap()
+
+   tms = gps.TMS.build(('s3://azavea-datahub/catalog', 'nlcd-tms-epsg3857'), display=cm)
+
+Of course, other color maps can be used.  See the documentation for
+:class:`~geopyspark.geotrellis.color.ColorMap` for more details.
+   
+:code:`TMS.build` can display data from catalogs—which are represented as a
+string-string pair containing the URI of the catalog root and the name of the
+layer—or from a :class:`~geopyspark.geotrellis.layer.Pyramid` object.  One may also
+specify a list of any combination of these sources; more on multiple sources
+below.
+
+Once a TMS server is constructed, we need to make the contents visible by
+binding the server.  The :meth:`tms.bind` method can take a ``host`` and/or a
+``port``, where the former is a string, and the latter is an integer.
+Providing neither will result in a TMS server accessible from localhost on a
+random port.  If the server should be accessible from the outside world, a
+``host`` value of ``"0.0.0.0"`` may be used.
+
+A call to :meth:`tms.bind` is then followed by a call to :meth:`tms.url_pattern`,
```

Comment:
Neither `tms.bind`, `tms.url_pattern`, nor `tms.unbind` produce a link to those respective methods. In order to produce links to the docs of these methods, they must be written this way:

```
:meth:`~geopyspark.geotrellis.tms.TMS.bind`
:meth:`~geopyspark.geotrellis.tms.TMS.unbind`
:meth:`~geopyspark.geotrellis.tms.TMS.uril_pattern`
```